src/util/include/util/DispatchPool.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Xournal++
+ *
+ * Template class for a dispatch pool
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <memory>
+#include <vector>
+
+namespace xoj::util {
+template <class T>
+class DispatchPool {
+public:
+    using listener_type = T;
+    DispatchPool() = default;
+    DispatchPool(DispatchPool&&) = default;
+    DispatchPool(const DispatchPool&) = default;
+    DispatchPool& operator=(DispatchPool&&) = default;
+    DispatchPool& operator=(const DispatchPool&) = default;
+
```

Comment:
Either make this class final *and* remove all default special member functions, or add a defaulted virtual destructor